### PR TITLE
Automation actions

### PR DESCRIPTION
**Pull request by RDMurray**
Hi.  This adds automation actions as discussed on the Wahtsapp group.

OSARA: Toggle global automation override between latch preview and off
OSARA: Report global / Track Automation Mode: Reports the global automation override if set, otherwise reports the automation mode of the current track
OSARA: Cycle automation mode of selected tracks: Cycles through automation modes for selected tracks.  This is needed because the sws action for cycling automation mode doesn't ensure that all selected tracks are set to the same mode.  

This depends on code in pr #216

### CODE REVIEW

**Comment by RDMurray** [comment]
Sorry, it looks like I messed this up as it contains the commit from #216 as well.let me know if it's a problem and I'll make another pr.

I didn't make any changes to the keymap as there is an on going discussion about it on WhatsApp

**Comment by jcsteh** [comment]
Thanks. This needs to be rebased against upstream master now that #216 is merged. You'll need to pull the current master branch from my repo and then rebase against it:
git rebase master
If you run into trouble, let me know and I'll make the change to your branch.

Also, this is now showing conflicts in osara.h. I'm not sure why.

**Comment by RDMurray** [comment]
Okay, I rebased it.

Thanks

Robbie

File: src/reaper_osara.cpp [hunk 1]
```diff
@@ -2055,6 +2055,47 @@ void cmdShowContextMenu3(Command* command) {
 	showReaperContextMenu(2);
 }
 
+void cmdReportAutomationMode(Command* command) {
+	// This reports the global automation override if set, otherwise tcurrent track automation mode.
```
Comment (jcsteh) [inline src/reaper_osara.cpp]: typo: tcurrent -> the current

File: src/reaper_osara.cpp [hunk 2]
```diff
@@ -2055,6 +2055,47 @@ void cmdShowContextMenu3(Command* command) {
...
+	}
+	outputMessage(s);
+}
+
+void cmdToggleGlobalAutomationLatchPreview(Command* command) {
+	if (GetGlobalAutomationOverride() == 5) {  // in latch preview mode
+		SetGlobalAutomationOverride(-1);
+		outputMessage("Global automation override off");
+	}
+	else { //not in latch preview.
```
Comment (jcsteh) [inline src/reaper_osara.cpp]: nit: Please put the else on the same line as the closing brace for the if; i.e. `} else {`

File: src/reaper_osara.cpp [hunk 3]
```diff
@@ -2055,6 +2055,47 @@ void cmdShowContextMenu3(Command* command) {
...
+	 		SetGlobalAutomationOverride(5);
```
Comment (jcsteh) [inline src/reaper_osara.cpp]: nit: Please fix indentation on this line.

File: src/reaper_osara.cpp [hunk 4]
```diff
@@ -2055,6 +2055,47 @@ void cmdShowContextMenu3(Command* command) {
...
+		outputMessage("Global automation override latch preview");
+	}
+}
+
+void cmdCycleTrackAutomation(Command* command) {
+	int count = CountSelectedTracks2(0, true);
+	if (count == 0)
+		return;
```
Comment (jcsteh) [inline src/reaper_osara.cpp]: It might be good to report "no selected tracks" or something here.

**Comment by RDMurray** [comment]
Okay no problem.  Derek suggested keeping this open until the key map changes are ready.  This might be a good idea considering that the readme has to be updated anyway.  What do you think?

**Pull request merged**